### ISSUE TITLE
Add searching

### ISSUE THREAD

**Issue by pR0Ps** [pR0Ps/website#3]
https://github.com/getpelican/pelican/issues/170

Options:
- Client Side
  - In general not scalable (user essentially has to download all content on the site)
  - Requires JS - not a great solution
  - Search the RSS feed
    - Use JS to parse the RSS feed and output results
    - See http://joevennix.com/2011/05/25/How-I-Implement-Static-Site-Search.html
    - RSS is done already by Pelican
    - RSS feed might not have all entries
    - RSS feed contains HTML, not bare data
  - Search a generated DB
    - Generate a JSON file with all the site data on it and use client-side JS to search it.
- Server-side
  - Scalable (user only has to download the results)
  - A bit of overhead from running the second application
  - See https://en.wikipedia.org/wiki/Fulltext#Free_and_open_source_software
  - See https://bitbucket.org/mchaput/whoosh/wiki/Home
  - API
    - A web application that searches the DB and returns the results, then use JS to output the results into the template.
    - Requires JS
  - Separate web application
    - Build another web application that searches, then uses the existing template to insert the content and return it to the user.
    - See https://github.com/MTecknology/pelican-search
    - No JS needed

**Comment by MTecknology** [comment]
I love your break down of different options. I'm excited to see I made the list!
Question... if I built the pelican-search system as a service, is it something you would use?

**Comment by pR0Ps** [comment]
If it was something I could run on my own server, then I would look into it, but I probably wouldn't use a service provided by another website.

**Comment by MTecknology** [comment]
Well... What I built is documented on github. You're free to do exactly
what I did.

**Comment by pR0Ps** [comment]
Yep, I've been looking into it, thanks! I just haven't gotten around to it yet...
